cypress/commands/api/alerts.ts

```diff
@@ -0,0 +1,157 @@
+// load the global Cypress types
+/// <reference types="cypress" />
+import { v1ApiPath, getCreds, saveIdOnly, makeAuthorizedRequest,makeAuthorizedRequestWithStatus } from "../server";
+import { logTestDescription, prettyLog } from "../descriptions";
+
+interface ComparableAlert {
+  id: number,
+  name: string,
+  frequencySeconds: number,
+  conditions: [{tag: string, automatic: boolean}],
+  lastAlert: boolean,
+  User: {
+	   id: number,
+	   username: string,
+	   email: string
+  },
+  Device: {
+	  id: number,
+	  devicename: string
+  }
```

Comment:
instead of parsing the id and devicename you could just pass the 'camera test name' here.   And convert it to an id at compare time.  This would get round a lot of the issues with timing and probably you wouldn't need some of the then statements.  

FYI Also when things  have id and name a layer down then that is just how our database code package works.   It basically means it is reading everything other than the id from the user table.   So we actually don't get much test benefit from checking these things in the test (especially every time).